peyvandak/Majlesi/django-docker-compose-url-shortener/urlshortener_api/api/views.py

```diff
@@ -0,0 +1,177 @@
+from .models import Shortener, VisitUrl
+from django.http import JsonResponse, HttpResponseRedirect, Http404
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login
+from django.contrib.auth import logout
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+
+@api_view(['POST'])
+def create_user(request):
+    if request.method == 'POST':
+        try:
+            user = User.objects.create_user(request.data['username'], request.data['email'], request.data['password'])
+            user.save()
+            return JsonResponse({"result": "new account has been created"})
+        except:
+            return JsonResponse({"result": 'Please enter "username" , "password", "email"'})
+
+
+@api_view(['GET'])
+def login_user(request):
+    if request.method == 'GET':
+        if request.user.is_authenticated:
+            return JsonResponse({"result": "You have logined before!"})
+        else:
+            user = authenticate(username=request.data["username"], password=request.data["password"])
+            if user is not None:
+                login(request, user)
+                result = "login successfully " + str(request.user.username) + "::"
+                return JsonResponse({"result": result}, safe=False)
+            else:
+                return JsonResponse({"result": "username or password is wrong. Try one more time."}, safe=False)
+
+
+@api_view(['GET'])
+def login_state(request):
+    if request.method == 'GET':
```

Comment:
I'm not sure but i think since you have used `@api_view(['GET'])`, you don't need to check `if request.method == 'GET'`
because `@api_view(['GET'])` would automatically send 405 response